### PR TITLE
Ensure public repositories in private organizations are visible and fix admin organizations list

### LINKED ISSUE #10144: Public repositories within private org are not visible to members
- Gitea version (or commit ref): 1.10.3 / 1.12.0+dev-251-gd4096ab6a
- Git version: Not relevant
- Operating system: Not relevant
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist: Not relevant

## Description

When repository, within an organization which has visibility set to Private, is set as public and user is part of a team that has no access specified to this repository, repository can be accessed via direct URI but can not be viewed on any list within UI.

I would expect that setting repository as public within Private organization would allow members of all teams to view it, regardless of team's permissions and team's repositories (for example I might want to make repository visible but read-only to all members of organization, while certain teams can write to different repositories).

This does not apply when organization is set as Limited. In such a case, repository is properly visible (to both members of organization as well as outside users).

Owners of org can of course see all repositories just fine.

## Screenshots

chrome_2020-02-05_01-32-09
chrome_2020-02-05_01-32-25
chrome_2020-02-05_01-32-38
chrome_2020-02-05_01-32-46
chrome_2020-02-05_01-32-52
firefox_2020-02-05_01-31-20
firefox_2020-02-05_01-31-33
firefox_2020-02-05_01-31-57

### PR DESCRIPTION
**Pull request by zeripath**
Although the #11442 fixed the accessibleRepositoryCondition, the collaboration condition also needed fixing. This PR fixes this.

Also fixes the organization list on the administration page.

Fix #10144

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: models/repo_list.go [hunk 1]
```diff
@@ -249,14 +249,35 @@ func SearchRepositoryCondition(opts *SearchRepoOptions) builder.Cond {
...
 				builder.Or(
-					builder.Expr("repository.id IN (SELECT repo_id FROM `access` WHERE access.user_id = ?)", opts.OwnerID),
-					builder.In("id", builder.Select("`team_repo`.repo_id").
+					// A. We have access
+					builder.In("`repository`.id",
+						builder.Select("`access`.repo_id").
+							From("access").
+							Where(builder.Eq{"`access`.user_id": opts.OwnerID})),
+					// B. We are in a team for
+					builder.In("`repository`.id", builder.Select("`team_repo`.repo_id").
```
Comment (guillep2k) [inline models/repo_list.go:264]: In theory, the `access` table covers the team membership case.

**Review by guillep2k — APPROVED** [review]
Just a remark. LG-TM.

File: models/repo_list.go (hunk 1 above)
Comment (zeripath) [reply models/repo_list.go:264]: it doesn't unfortunately otherwise none of this code would be necessary...

File: models/repo_list.go [hunk 2]
```diff
@@ -249,14 +249,35 @@ func SearchRepositoryCondition(opts *SearchRepoOptions) builder.Cond {
...
-						Join("INNER", "team_user", "`team_user`.team_id = `team_repo`.team_id"))),
-				builder.Neq{"owner_id": opts.OwnerID})
+						Join("INNER", "team_user", "`team_user`.team_id = `team_repo`.team_id")),
+					// C. Public repositories in private organizations that we are member of
+					builder.And(
+						builder.Eq{"`repository`.is_private": false},
+						builder.In("`repository`.owner_id",
+							builder.Select("`org_user`.org_id").
+								From("org_user").
+								Join("INNER", "user", "`user`.id = `org_user`.org_id").
```
Comment (zeripath) [inline models/repo_list.go]:
```suggestion
								Join("INNER", "`user`", "`user`.id = `org_user`.org_id").
```

**Comment by zeripath** [comment]
Make lg-tm work

**Comment by 6543** [comment]
backport done :)

**Pull request merged**